frontends/qt/server/server.go

```diff
@@ -91,6 +92,78 @@ func handleURI(uri *C.char) {
 	bridgecommon.HandleURI(C.GoString(uri))
 }
 
+func matchDarkTheme(themeName string) bool {
+	return strings.Contains(strings.ToLower(strings.TrimSpace(themeName)), "dark")
+}
+
+// detect theme used by OS and return true if it's dark
+func detectDarkTheme() bool {
+	log := logging.Get().WithGroup("server")
+	switch myos := strings.ToLower(runtime.GOOS); myos {
+	case "darwin":
+		cmd := exec.Command("defaults", "read", "-g", "AppleInterfaceStyle")
+		out, err := cmd.Output()
+		if err == nil {
+			log.Info("MacOS theme: " + string(out))
+			if strings.TrimSpace(string(out)) == "Dark" {
+				return true
+			}
+		}
+	case "linux":
```

Comment:
i couldn't test this as my qubes system does not patch through this system setting to the vm's.